in_toto/in_toto_sign.py

```diff
@@ -0,0 +1,265 @@
+    sign -r -i  /bob/software/in-toto/test/package.link
+
+"""
+import os
+import sys
+import argparse
+import in_toto.log
+import in_toto.util
+from in_toto.models.common import Signable as signable_object
+from in_toto.models.link import Link as link_import
+import securesystemslib.exceptions
+import securesystemslib.keys
+import securesystemslib.formats
+
+
+def add_sign(link, key):
+    """
+    <Purpose>
+      Signs the given link file with the corresponding key,
+      adds the signature to the file, and then returns, the
+      link file as an object.
+
+    <Arguments>
+      link - path to the signable link file
+      key - the key to be used for signing
+
+    <Exceptions>
+      None
+
+    <Returns>
+      An object containing the contents of the link file after
+      adding the signature
+
+    """
+    # reading the file from the given link and making an object
+    signable_object = link_import.read_from_file(link)
+
+    rsa_key = in_toto.util.import_rsa_key_from_file(key)
+
+    # checking if the given key follows the format
+    securesystemslib.formats.KEY_SCHEMA.check_match(rsa_key)
+
+    # create the signature using the key, and the link file
+    signature = securesystemslib.keys.create_signature(rsa_key, signable_object.payload)
+
+    # append the signature into the file
+    signable_object.signatures.append(signature)
+
+    return signable_object
+
+
+def add_replace(link, key):
+    """
+    <Purpose>
+      Replaces all the exisiting signature with the new signature,
+      signs the file, and then returns the link file as an object.
+
+    <Arguments>
+      link - path to the key file
+      key - the key to be used for signing
+
+    <Exceptions>
+      None
+
+    <Returns>
+      An object containing the contents of the link file after
+      adding the signature which replaces the old signatures
+    """
+
+    # reading the link file and making an object
+    signable_object = link_import.read_from_file(link)
+
+    # import rsa key from the filepath
+    rsa_key = in_toto.util.import_rsa_key_from_file(key)
+
+    # check if the key corresponds to the correct format
+    securesystemslib.formats.KEY_SCHEMA.check_match(rsa_key)
+
+    # core working
+    signature = securesystemslib.keys.create_signature(rsa_key, signable_object.payload)
+    signable_object.signatures = []
+    signable_object.signatures.append(signature)
+
+    return signable_object
+
+
+def verify_sign(link, key_pub):
+    """
+    <Purpose>
+      Verifies the signature field in the link file, given a public key
+
+    <Arguments>
+      link - path to the link file
+      key_pub - public key to be used to verification
+
+    <Exceptions>
+      Raises SignatureVerificationError
+        - 'Invalid Signature' : when the verification fails
+        - 'Signature Key Not Found' : when KeyError occurs
+        - 'No Signatures Found' - when no signature exists
+
+    <Returns>
+      Boolean True when the verification is success
```

Comment:
Alright! Since I am using link.verify_signatures(key_dict), in the function verify_signatures, exceptions are already raised. So what I can do is that, either it returns nothing, the verification was successful, or it catches the exception which occurred while running and prints it . What do you say?